app/src/main/java/com/wednesday/banklocator/adapter/IfscAdapter.kt

```diff
@@ -0,0 +1,64 @@
+package com.wednesday.banklocator.adapter
+
+import android.view.Gravity
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.Toast
+import androidx.recyclerview.widget.RecyclerView
+import com.wednesday.banklocator.R
+import com.wednesday.banklocator.model.IfscResponse
+import com.wednesday.banklocator.viewmodel.BankViewModel
+import kotlinx.android.synthetic.main.partial_banks_list.view.*
+
+class IfscAdapter(var item: IfscResponse, private val viewModel: BankViewModel) :
```

Comment:
I don't think it makes sense to have two adapters one for the network response and another for the database.

At the end of the day the data that you're showing is the same whether the source is the database or network. You just need a way to identify the source which could be a part of your data item i.e. the base data class.

Reply:
The reason we're using two adapters is that the information from BANK API is coming in a single item and the information in the Local Database is stored in the form of a list. That is why we're using two.